app/Http/Controllers/Front/HomeController.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace App\Http\Controllers\Front;
+
+use App\Http\Controllers\Front\BaseController;
+use Illuminate\Http\Request;
+
+class HomeController extends BaseController
+{
+    public function index() {
+    	if(!isset($this->globaldata['user'])){
+			return view('welcome');
+    	}
+    	else {
+    		return redirect()->to('posts');
+    	}
```

Comment:
Global data can sometimes be hard to reason about and there are numerous conventions in Laravel that would be better here - see [Laravel docs](https://laravel.com/docs/master/authentication) 

You could just use `Auth::check()`